test/ProblemDetailsResponseFactoryTest.php

```diff
@@ -176,4 +178,14 @@ public function testFactoryRendersPreviousExceptionsInDebugMode() : void
         $this->assertEquals(101010, $payload['exception']['stack'][0]['code']);
         $this->assertEquals('first', $payload['exception']['stack'][0]['message']);
     }
+
+    public function testFragileDataInExceptionMessageShouldBeHiddenInResponseBodyInNoneDebugMode()
+    {
+        $fragileMessage = 'Your SQL or password here';
+        $exception = new \Exception($fragileMessage);
+
+        $response = $this->factory->createResponseFromThrowable($this->request->reveal(), $exception);
+
+        $this->assertNotContains($fragileMessage, (string) $response->getBody());
```

Comment:
If we want to provide details you should use `ProblemDetailsException`. User should see only details of 4xx. 5xx are internal and api developer should only know about this. For others should be `Internal server error` provided or custom `ProblemDetailsException` if we really want to say something.

Show exception details out of box is dangerous and we should care about this in most common cases by serve safe message.